pallets/liquid-staking-two/src/lib.rs

```diff
@@ -0,0 +1,607 @@
+                    who,
+                    &era_index,
+                    &current_era,
+                    &user_ledger_per_era,
+                    &pool_ledger_per_era,
+                ),
+                StakingOperationType::Matching => user_ledger_per_era.remaining_withdrawal_limit(), //if matching, can claim all directly
+                _ => (0, 0),
+            };
+
+            MatchingQueueByUser::<T>::mutate(who, &era_index, |b| {
+                b.claimed_unstake_amount = b
+                    .claimed_unstake_amount
+                    .saturating_add(claim_unstake_each_era);
+                b.claimed_stake_amount =
+                    b.claimed_stake_amount.saturating_add(claim_stake_each_era);
+
+                if b.total_stake_amount == b.claimed_stake_amount
+                    && b.total_unstake_amount == b.claimed_unstake_amount
+                {
+                    // user have already claimed all he can claim in this era, remove it from MatchingQueue
+                    remove_record_from_user_queue.push(era_index);
+                }
+            });
+
+            *withdrawable_unstake_amount =
+                withdrawable_unstake_amount.saturating_add(claim_unstake_each_era);
+            *withdrawable_stake_amount =
+                withdrawable_stake_amount.saturating_add(claim_stake_each_era);
+        }
+    }
+
+    // if bond, need to wait 1 era or get the matching part instantly
+    // users who unstake can get KSM back directly, because this era is about bond operation,
+    // so all user who unstake in this era, no need wait 28 eras.
+    // after waiting 1 era, users who stake get the left part
+    fn claim_in_bond_operation(
+        who: &T::AccountId,
+        claim_era: &EraIndex,
+        current_era: &EraIndex,
+        user_ledger_per_era: &UserLedgerPerEra,
+        pool_ledger_per_era: &PoolLedgerPerEra,
+    ) -> WithdrawalAmount {
+        if claim_era.clone() + T::StakingDuration::get() > current_era.clone() {
+            if !user_ledger_per_era.claimed_matching {
+                MatchingQueueByUser::<T>::mutate(who, claim_era, |b| {
+                    b.claimed_matching = true;
+                });
+
+                user_ledger_per_era.instant_withdrawal_by_bond(pool_ledger_per_era)
+            } else {
+                (0, 0)
+            }
+        } else {
+            user_ledger_per_era.remaining_withdrawal_limit()
+        }
+    }
+
+    // if unbond, normally need to wait 28 eras
+    // after waiting 28 eras, users who unstake and get the left part
+    // considering users who stake and forget to claim within 28 eras.
+    fn claim_in_unbond_operation(
+        who: &T::AccountId,
+        claim_era: &EraIndex,
+        current_era: &EraIndex,
+        user_ledger_per_era: &UserLedgerPerEra,
+        pool_ledger_per_era: &PoolLedgerPerEra,
+    ) -> WithdrawalAmount {
+        if claim_era.clone() + T::BondingDuration::get() > current_era.clone() {
```

Comment:
even if the previous advice doesn't work , you can still remove these two clone, because `u32` implemented `Copy`

Reply:
Much complicated. They're declared as reference in parameters. I've changed them passed by value.